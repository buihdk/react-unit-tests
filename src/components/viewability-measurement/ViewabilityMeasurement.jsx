import React, { memo, useState } from 'react';
import PropTypes from 'prop-types';
import I18n from 'i18n-js';
import { Form, Row, Input, Alert } from 'antd';

import CustomAlert from 'src/components/alert/CustomAlert';

import {
  getArrayValue,
  useValidateLengths,
} from './ViewabilityMeasurement.utils';

const styles = {};

const { Item } = Form;
const { TextArea } = Input;
const formItemLayout = { labelCol: { span: 24 }, wrapperCol: { span: 24 } };

const ViewabilityMeasurement = ({
  form: { getFieldDecorator, getFieldValue },
  measurementJSURLs,
  measurementVendorKeys,
  measurementParams,
}) => {
  const [hasError, setHasError] = useState(false);
  const arr1 = getArrayValue(getFieldValue('measurementJSURLs'));
  const arr2 = getArrayValue(getFieldValue('measurementVendorKeys'));
  const arr3 = getArrayValue(getFieldValue('measurementParams'));

  getFieldDecorator('equalMeasurementLengths', {
    rules: [{ required: hasError }],
  });
  useValidateLengths({ arr1, arr2, arr3, setHasError });

  return (
    <Row>
      <CustomAlert
        message={I18n.t('measurement.title')}
        description={I18n.t('measurement.description')}
      />
      <Alert
        message={I18n.t('messages.inputOneItemPerLine')}
        className={styles.warning}
        type="warning"
        showIcon
      />
      <Item {...formItemLayout} label={I18n.t('measurement.measurementJSURLs')}>
        {getFieldDecorator('measurementJSURLs', {
          initialValue: measurementJSURLs,
          rules: [
            {
              type: 'string',
              whitespace: true,
              message: I18n.t('messages.wrongStringFormat'),
            },
          ],
        })(<TextArea rows={4} />)}
      </Item>
      <Item
        {...formItemLayout}
        label={I18n.t('measurement.measurementVendorKeys')}
      >
        {getFieldDecorator('measurementVendorKeys', {
          initialValue: measurementVendorKeys,
          rules: [
            {
              type: 'string',
              whitespace: true,
              message: I18n.t('messages.wrongStringFormat'),
            },
          ],
        })(<TextArea rows={4} />)}
      </Item>
      <Item {...formItemLayout} label={I18n.t('measurement.measurementParams')}>
        {getFieldDecorator('measurementParams', {
          initialValue: measurementParams,
          rules: [
            {
              type: 'string',
              whitespace: true,
              message: I18n.t('messages.wrongStringFormat'),
            },
          ],
        })(<TextArea rows={4} />)}
      </Item>
      <Item
        validateStatus="error"
        className={styles[!hasError && 'hideMsg']}
        help={I18n.t('measurement.equalMeasurementLengthsErrorMsg')}
      />
    </Row>
  );
};

ViewabilityMeasurement.propTypes = {
  form: PropTypes.shape({
    getFieldDecorator: PropTypes.func,
    getFieldValue: PropTypes.func,
    getFieldError: PropTypes.func,
  }).isRequired,
  measurementJSURLs: PropTypes.string,
  measurementVendorKeys: PropTypes.string,
  measurementParams: PropTypes.string,
};

ViewabilityMeasurement.defaultProps = {
  measurementJSURLs: null,
  measurementVendorKeys: null,
  measurementParams: null,
};

export default memo(ViewabilityMeasurement);
