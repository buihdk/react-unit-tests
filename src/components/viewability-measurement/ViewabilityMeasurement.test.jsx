import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ViewabilityMeasurement from './ViewabilityMeasurement';

const form = {
  getFieldDecorator: () => element => element,
  getFieldValue: jest.fn(),
  getFieldError: jest.fn(),
};

describe('ViewabilityMeasurement', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<ViewabilityMeasurement form={form} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
