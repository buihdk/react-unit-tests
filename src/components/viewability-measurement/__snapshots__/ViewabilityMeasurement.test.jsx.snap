// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewabilityMeasurement renders without crashing 1`] = `
<Row>
  <CustomAlert
    className=""
    description="[missing \\"en.measurement.description\\" translation]"
    message="[missing \\"en.measurement.title\\" translation]"
  />
  <Alert
    message="[missing \\"en.messages.inputOneItemPerLine\\" translation]"
    showIcon={true}
    type="warning"
  />
  <FormItem
    label="[missing \\"en.measurement.measurementJSURLs\\" translation]"
    labelCol={
      Object {
        "span": 24,
      }
    }
    wrapperCol={
      Object {
        "span": 24,
      }
    }
  >
    <TextArea
      rows={4}
    />
  </FormItem>
  <FormItem
    label="[missing \\"en.measurement.measurementVendorKeys\\" translation]"
    labelCol={
      Object {
        "span": 24,
      }
    }
    wrapperCol={
      Object {
        "span": 24,
      }
    }
  >
    <TextArea
      rows={4}
    />
  </FormItem>
  <FormItem
    label="[missing \\"en.measurement.measurementParams\\" translation]"
    labelCol={
      Object {
        "span": 24,
      }
    }
    wrapperCol={
      Object {
        "span": 24,
      }
    }
  >
    <TextArea
      rows={4}
    />
  </FormItem>
  <FormItem
    help="[missing \\"en.measurement.equalMeasurementLengthsErrorMsg\\" translation]"
    validateStatus="error"
  />
</Row>
`;
